lib/application.js

```diff
@@ -296,7 +296,9 @@ app.enableAuth = function() {
     var req = ctx.req;
     var Model = method.ctor;
     var modelInstance = ctx.instance;
-    var modelId = modelInstance && modelInstance.id || req.param('id');
+    var modelId = (modelInstance && modelInstance.id) ||
+      (req.params && req.params.id) ||
+      (req.body && req.body.id);
```

Comment:
> but the id should not be in the request body

That's AFAIK not true, e.g. `updateOrCreate` takes the id from the request body.

More importantly, we must preserve backwards compatibility with previous loopback versions. So even if the `id` was not supposed to be read from the body, I would argue that we should still check the body for the sake of staying backwards compatible.